photon-lib/src/main/scala/com/linkedin/photon/ml/function/PriorDistribution.scala

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2019 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.function
+
+import breeze.linalg.{DenseMatrix, DenseVector, Vector, diag}
+import breeze.numerics.sqrt
+import com.linkedin.photon.ml.constants.MathConst
+import com.linkedin.photon.ml.normalization.NormalizationContext
+import com.linkedin.photon.ml.model.{Coefficients => ModelCoefficients}
+import com.linkedin.photon.ml.util.BroadcastWrapper
+
+/**
+ * Trait for an incremental training objective function. It is assumed that the prior is a product of Gaussian and
+ * Laplace distributions. The L1 regularization weight refers to the relative weight of the Laplace prior. The L2
+ * regularization weight refers to the relative weight of the Gaussian prior.
+ */
+trait PriorDistribution extends ObjectiveFunction {
+
+  val priorCoefficients: ModelCoefficients = ModelCoefficients(DenseVector.zeros(1))
+
+  lazy protected val priorMeans: Vector[Double] = priorCoefficients.means
+  lazy protected val priorVariances: Vector[Double] = priorCoefficients.variancesOption.get
+  lazy protected val inversePriorVariances: DenseVector[Double] = priorVariances.map(v => if (v > MathConst.EPSILON) 1.0 / v else l2RegWeight).toDenseVector
+  protected var l2RegWeight: Double = 0D
+  protected var incrementalWeight: Double = 1D
+
+  require(l2RegWeight >= 0D, s"Invalid regularization weight '$l2RegWeight")
+
+  /**
+   * Getter for the Gaussian weight of the prior.
+   *
+   * @return The L2 regularization weight
+   */
+  def l2RegularizationWeight: Double = incrementalWeight
```

Comment:
Very confusing - a function called `l2RegularizationWeight` which returns `incrementalWeight` in an object which has another member called `l2RegWeight`.

Reply:
fixed